### PR TITLE
add rpm_ostree and fact info

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <admiller@redhat.com>

##### SUMMARY

rpm-ostree based modules specifically for RHEL 4 Edge

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

r4e_rpm_ostree.py
rpm_ostree_upgrade.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Akasurde** [comment]
@maxamillion `rpm_ostree_pkg` already exists in [`community.general`](https://github.com/ansible-collections/community.general/blob/main/plugins/modules/packaging/os/rpm_ostree_pkg.py).

**Comment by jborean93** [comment]
I don't believe we are accepting any new modules and these should be in a collection rather than here. There's `ansible.posix` and `community.general` which is generally where adhoc stuff is directed to.

File: lib/ansible/module_utils/facts/system/pkg_mgr.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
             {'path': '/usr/sbin/sorcery', 'name': 'sorcery'},
             {'path': '/usr/bin/installp', 'name': 'installp'},
             {'path': '/QOpenSys/pkgs/bin/yum', 'name': 'yum'},
+            {'path': '/usr/bin/rpm-ostree', 'name': 'rpm_ostree'},
```
Comment (jborean93) [inline lib/ansible/module_utils/facts/system/pkg_mgr.py]: This won't work as `/usr/bin/rpm-ostree` has an entry at the start for `atomic_container`.

**Comment by bcoca** [comment]
just a note, we are not taking new modules into core (the fact update should be fine though), I would suggest the ansible.posix collection for the modules.

File: lib/ansible/modules/r4e_rpm_ostree.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                pkgs.append(pkg)
+    elif module.params['state'] in ['absent', 'removed']:
+        for pkg in module.params['name']:
+            if locally_installed(module, pkg):
+                pkgs.append(pkg)
+
+    if not pkgs:
+        module.exit_json(msg="No changes made.")
+    else:
+        module.fail_json(msg="The following packages are absent in the currently booted rpm-ostree commit: %s" ' '.join(pkgs))
```
Comment (Akasurde) [inline lib/ansible/modules/r4e_rpm_ostree.py]:
```suggestion
        module.fail_json(msg="The following packages are absent in the currently booted rpm-ostree commit: %s" % ' '.join(pkgs))
```

**Pull request closed without merging**